server/sonar-server/src/main/java/org/sonar/server/es/EsClient.java

```diff
@@ -212,17 +208,19 @@ public long getMaxFieldValue(String indexName, String typeName, String fieldName
   public void start() {
     if (nativeClient == null) {
       ESLoggerFactory.setDefaultFactory(new Slf4jESLoggerFactory());
-      org.elasticsearch.common.settings.Settings esSettings = ImmutableSettings.settingsBuilder()
-        .put("node.name", StringUtils.defaultIfEmpty(settings.getString(ProcessProperties.CLUSTER_NODE_NAME), "sq_local_client"))
-        .put("network.bind_host", StringUtils.defaultIfEmpty(settings.getString(ProcessProperties.SEARCH_HOST), "localhost"))
-        .put("node.rack_id", StringUtils.defaultIfEmpty(settings.getString(ProcessProperties.CLUSTER_NODE_NAME), "unknown"))
+      org.elasticsearch.common.settings.Settings esSettings = org.elasticsearch.common.settings.Settings.builder()
+        .put("node.name", defaultIfEmpty(settings.getString(ProcessProperties.CLUSTER_NODE_NAME), "sq_local_client"))
+        .put("network.bind_host", defaultIfEmpty(settings.getString(ProcessProperties.SEARCH_HOST), "localhost"))
+        .put("node.rack_id", defaultIfEmpty(settings.getString(ProcessProperties.CLUSTER_NODE_NAME), "unknown"))
         .put("cluster.name", StringUtils.defaultIfBlank(settings.getString(ProcessProperties.CLUSTER_NAME), "sonarqube"))
         .build();
-      nativeClient = new TransportClient(esSettings);
-      ((TransportClient) nativeClient).addTransportAddress(new InetSocketTransportAddress(StringUtils.defaultIfEmpty(settings.getString(ProcessProperties.SEARCH_HOST),
-        LoopbackAddress.get()
-          .getHostAddress()),
-        settings.getInt(ProcessProperties.SEARCH_PORT)));
+      nativeClient = TransportClient.builder().settings(esSettings).build();
+      String host = defaultIfEmpty(settings.getString(ProcessProperties.SEARCH_HOST), LoopbackAddress.get().getHostAddress());
+      try {
+        ((TransportClient) nativeClient).addTransportAddress(new InetSocketTransportAddress(InetAddress.getByName(host), settings.getInt(ProcessProperties.SEARCH_PORT)));
+      } catch (UnknownHostException e) {
+        // TODO
```

Comment:
![INFO](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png) Complete the task associated to this TODO comment. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS1135)